### DISCUSSION TITLE
What do you think about my Video extension?

### DISCUSSION THREAD

**Discussion by GiovanniSalmeri** [datenstrom/community#851] [Ask a question] [answered]
Hello everybody. I have made a new little [Video extension](https://github.com/GiovanniSalmeri/yellow-video). It is based on the [YouTube extension](https://github.com/annaesvensson/yellow-youtube) and acts as a drop-in replacement for it, but it accepts as video sources also local videos, remote videos (mp4, ogg or webm) and videos from 14 different services. The service is recognised with the help of a little regex magic. I thought of this extension principally for PeerTube instances (in order to be a bit freer from the Google embrace), but later I realised that it was very easy to extend it to other services. I intentionally excluded those which share gravely objectionable contents from a legal point of view.

Any comments are welcome. Thank you!

**Comment by PetersOtto** [comment]
Hello Giovanni.
Thanks for the extension! 
I like embedded videos but regarding privacy I prefer a direct link to Youtube or other services.
I would be interested to know how you handle third party cookies.

**Reply by GiovanniSalmeri** [reply]
Hello! As regards third-party cookies, presently I am not aware of any method an extension could use in order to influence their use. In my daily use, I simply exclude them with the settings of the browser. Moreover, I use a very simple Linux-script in order to update my `hosts` file, and so I am always free enough from ads, trackings and nasty sites, without the need of any browser add-on.

A direct link to Youtube and other services does seem to me a step ahead, because in this way every user is free to follow or not to follow the link, but it is also true that the problem is only delayed... The main purpose of my extension is actually to offer an alternative to Youtube: with local or remote video files (no cookies), or with services (like Peertube) more respectful of privacy, or at least with services *different* from the Google owned Youtube. The less Google (and Microsoft, and Facebook/Meta) is under my eyes, the more I am happy :-)

**Reply by GiovanniSalmeri** [reply]
Well, since my previous answer I have learned that something could actually be tried in the extension itself: see for example this answer in [Stackoverflow](https://stackoverflow.com/questions/59291907/is-it-possible-to-block-cookies-within-an-iframe). I will try and possibly update the extension!

**Reply by PetersOtto** [reply]
Hello Giovanni!
Thank you for your explanation! 
When I create websites, I try to get along without cookies. I do not want the customer to have problems with the website. In my opinion the easiest way to make a website GDPR compliant is to not use cookies. 

A good GDPR cookie consent that prevents the loading of all third-party content (before users consent), I have not yet found.

Therefore I asked.

**Reply by GiovanniSalmeri** [reply]
> Well, since my previous answer I have learned that something could actually be tried in the extension itself: see for example this answer in [Stackoverflow](https://stackoverflow.com/questions/59291907/is-it-possible-to-block-cookies-within-an-iframe). I will try and possibly update the extension!

Follow-up: Yes, it is possible to block third-party cookies in the extension, but in this case many a service (among which Youtube) will prevent the video to play. So it seems to me futile to block cookies in the extension: if one does not want them, one must avoid services that rely on them.

**Comment by markseuffert** [comment, marked as answer]
Thank you for the extension. I like the idea of self-hosted video files and it appears you are already many steps ahead. :smile: 

About the shortcut arguments. Perhaps we can differentiate better between local files and streaming services. Where for the later you specify the URL. Yes, the extension would no longer be a drop-in replacement, but I think other extensions should also use an URL for external services. Would also make for a better user experience. Just a technical idea.... stumbled upon while doing some spring cleaning.

**Reply by GiovanniSalmeri** [reply]
I agree! After a little reflection, I separated the logic for streaming services from this extension and from [Sound](https://github.com/GiovanniSalmeri/yellow-sound) and I made a different extension [Embed](https://github.com/GiovanniSalmeri/yellow-embed). I also dropped the regex magic which I used in both extensions: it was fun, but also fragile and opaque (from the markdown source one couldn't know at a glance which service was being used!). Finally, I moved the definition of the services in an `.ini` file. <del>Obviously I will drop as soon as possible the logic for streaming services from Video and Sound, which will be only for self-hosted files.</del>

I have dropped the support for streaming services in the [Video](https://github.com/GiovanniSalmeri/yellow-video) extension. As soon as possible I will add the support for subtitles, which is relevant for accessibility and usability.

**Reply by markseuffert** [reply]
By the way, at the moment only MP4 and OGG are supported in the editor. 

Perhaps use `video.mp4` or `mandalorian.mp4` in the examples?

https://user-images.githubusercontent.com/2429165/160802010-05f6808c-670f-4415-80e4-131114aa164c.mp4
